src/Compilers/CSharp/Portable/BoundTree/BoundDagTest.cs

```diff
@@ -39,5 +43,29 @@ public override int GetHashCode()
         {
             return Hash.Combine(Kind.GetHashCode(), Input.GetHashCode());
         }
+
+#if DEBUG
+        private int _id;
+        private bool _idWasRead;
+
+        public int Id
+        {
+            get
+            {
+                _idWasRead = true;
+                return _id;
+            }
+            internal set
+            {
+                Debug.Assert(!_idWasRead, "Id was set after reading it");
+                _id = value;
+            }
+        }
+
+        internal new string GetDebuggerDisplay()
+        {
+            return "t" + Id;
```

Comment:
Maybe use `_` underscore to display uninitialized value? `t_`?

Reply:
Done